id-repository/id-repository-core/src/main/java/io/mosip/idrepository/core/builder/IdentityIssuanceProfileBuilder.java

```diff
@@ -45,36 +52,54 @@
 	private static String dateFormat;
 
 	public IdentityIssuanceProfile build() {
-		if (StringUtils.isBlank(filterLanguage))
-			filterLanguage = this.getPreferredLanguage(oldIdentity);
-		buildOldProfile();
-		buildNewProfile();
-		IdentityIssuanceProfile profile = new IdentityIssuanceProfile();
-		profile.setProcessName(processName);
-		profile.setDate(LocalDate.now());
-		profile.setOldProfile(oldProfile);
-		profile.setNewProfile(newProfile);
-		return profile;
+		try {
+			if (StringUtils.isBlank(filterLanguage))
+				IdentityIssuanceProfileBuilder.setFilterLanguage(this.getPreferredLanguage(newIdentity));
```

Comment:
Setting preferred language statically may not be appropriate for a multiple parellel request processing.

Reply:
Updated to class variable